### DISCUSSION TITLE
RSECon 2024 session: Weather & climate RSE community

### DISCUSSION THREAD

**Discussion by jmarshrossney** [NERC-CEH/rse_group#31] [General]
We (RSE group) have been invited to a [session at RSECon](https://virtual.oxfordabstracts.com/#/event/49081/submission/40) which aims to "bring together Climate and Weather RSEs to learn from one another, network to form lasting partnerships, and discuss whether and what we are missing as a community."

### Preparation for our part of the presentation

Part of the session is a presentation in which RSE groups from different organisations can briefly outline what they do. I'm putting together our slides for this part, which are based on the following:

1. What does UKCEH do in the weather & climate space? (with emphasis on things that RSEs are/will be involved with)
2. How do our RSEs work?
3. What tools and approaches do we use, and why? (Mention our backgrounds if useful)
4. What do we want or need from the weather & climate community?

I have 2 slides and probably just a couple of minutes to go through this, so I'm not exactly short of content, but **if there is anything you feel is particularly important to mention here please let me know!** Particularly if you want to put a certain topic on the agenda for the wider discussion that will follow.

### Other topics for discussion

The organisers pose some questions in [their abstract](https://virtual.oxfordabstracts.com/#/event/49081/submission/40). 

> Do we want to form a Special Interest Group under the RSE Society?

and

> Do we want regular meetings or seminars? Or do we just need signposting to existing ones?

I think it's perfectly fine for us to turn up to the session without strong priors on these topics - after all we are a brand new RSE group and UKCEH is for the most part a consumer of weather and climate science and data rather than a producer. That said, **if anyone has thoughts on these questions we could start this discussion here.**

### After the session

I will share my notes from the session on this discussion thread.

I wonder if this could be a blueprint for how we engage *as a group* with events (workshops, seminars etc.) in future, when typically only one or two of us will actually attend the event. We could make a dedicated category and template for discussions around events.

**Comment by jmarshrossney** [comment]
I got this far earlier today. Could do with some input on (3) especially.

> 1. What does UKCEH do in the weather & climate space? (with emphasis on things that RSEs are/will be involved with)

- UKCEH is a significant *consumer* of weather and climate data, for research areas include land-surface science, pollution and atmospheric composition, hydro-climate modelling, ecological modelling, others...?
- A flagship project is JULES (Joint UK Land Environment Simulator) which is the terrestrial biogeochemistry component of the UK Earth System Model, coupling to the [Met Office's Unified Model](https://www.metoffice.gov.uk/research/approach/modelling-systems/unified-model/index)
	- UKCEH is the UK academic lead for this model
	- JULES provides land-surface boundary conditions for climate modelling via the Unified Model
	- But it also consumes weather and climate data to constrain those same boundaries during explicit land surface modelling
- I have a meeting with members of the ACC group next Tuesday, so will hopefully gain some understanding of if/how they use weather & climate science outputs
- @mattjbr123, anything spring to mind on the hydrology side of things? What was the experience like implementing atmospheric/oceanic boundary conditions in [Unifhy](https://github.com/unifhy-org/unifhy)?


> 2. How do our RSEs work?

- 6 RSEs (3 senior), embedded in the (large) Environmental Data Science group which includes SEs, data scientists, systems architects, UX engineers, social scientists...
- New team - we all joined in summer 2024
- Based in 3/4 locations around the UK, so most communication is based on Teams & GitHub
- Open bidding process for O(months) of 1-2 RSE time
	- One RSE typically works on 2 projects at any given time
- Aiming to have RSEs playing a key role in fostering a community of practice so that scientists can get help with the day-to-day as well
- Training is also a key aspect of our role
- Aiming to be almost maximally open with our communication - use GitHub discussions etc.
	- Actively discussing how we can work "in the open" in a way that makes it easier for others to follow, dip in and out, contribute. That doesn't necessarily mean publish the minutes of every meeting, but it could - we need to figure this out

> 3. What tools and approaches do we use, and why? (Mention our backgrounds if useful)

- We have a new [digital strategy](https://www.ceh.ac.uk/about-us/digital-strategy) which determines what we work on and somewhat informs how we work
	- [The Turing Way](https://www.turing.ac.uk/research/research-projects/turing-way)
	- [The FAIR Guiding Principles for scientific data management and stewardship](https://www.nature.com/articles/sdata201618)
- We are working on a [Team agreement](https://github.com/NERC-CEH/rse_group/discussions/23)
	- Idea is to have some red lines, but also some things that we robustly advocate for but ultimately meet scientists somewhere in the middle
	- We are supposed to be "more than" just IT support - we bring our own research experience to the project - often the problem they scientists think they want solving is not really the problem that they want solving
	- But ultimately we are there to facilitate better science - science which has been progressing happily without us up to now. It's possible to overstate our importance!
- Scientists here often work with Python/R notebooks through DataLabs

>4. What do we want or need from the weather & climate community?

- Big caveat: I am new to the field, so my opinions are not close to being fully formed
	- Which is why I need to talk to other users of w&c science in UKCEH
- Being part of a community of practice, sharing skills and knowledge, finding opportunities for alignment and collaboration
- Better 'modernised' documentation and examples for climate models
	- Ideally make it easier to run small simulations locally
	- Why not use GitHub and open-source? 
	- Make it easier to have open discussions and flag issues?

**Reply by mattjbr123** [reply]
Many thanks for putting this together @jmarshrossney 
Just a few things I could think of to add, but otherwise very comprehensive and like you say this is only a brief 2min intro so no need to add tons of material!

> * @mattjbr123, anything spring to mind on the hydrology side of things? What was the experience like implementing atmospheric/oceanic boundary conditions in [Unifhy](https://github.com/unifhy-org/unifhy)?

Yeah I think it's a similar picture, at least for the modelling that I'm familiar with, weather and climate datasets are used as the boundary conditions. For UniFHy specifically, as it's a framework and not the actual models themselves, it has the requirement for boundary conditions to be specified in a particular format, which is onerous as it requires lots of wrangling. Eventually the idea would be to have the atmosphere as a component in the framework itself, i.e. make it dynamically modelled rather than boundary conditions. 



> * UKCEH is a significant _consumer_ of weather and climate data, for research areas include land-surface science, pollution and atmospheric composition, hydro-climate modelling, ecological modelling, others...?

Crop modelling too. I worked on the [CropNet](https://github.com/OpenCLIM/cropnet) and [EcoCrop](https://github.com/OpenCLIM/ecocrop) (part of OpenCLIM) projects before moving to the RSE role. These used climate data as the driving data too. 



> Better 'modernised' documentation and examples for climate models
> 
>     * Ideally make it easier to run small simulations locally
> 
>     * Why not use GitHub and open-source?
> 
>     * Make it easier to have open discussions and flag issues?

Could also add:
- Make interoperability of the various models better (like what UniFHy is ultimately trying to acheive)
